az-core/src/test/java/azkaban/utils/ExecutorServiceUtilsTest.java

```diff
@@ -80,7 +81,7 @@ public void can_not_submit_tasks_after_shutdown() throws
 
   private void sleep() {
     try {
-      Thread.sleep(1000);
+      Thread.sleep(30_000);
```

Comment:
30_000 sounds too long here. What if the shutdown method fail? Will it take 30 seconds to fail this particular test? I'd like this combination: Thread.sleep(5000) and islessThan(1000).

If JVM can not interrupt the sleep thread within 1s, there must be something seriously wrong.

Reply:
I don't think the change would make any difference. If the shutdown method does nothing, the test will still run as quickly as the host allows. 

Thanks for your comment, I realized that the test is not testing the right thing. Made a change.

Reply:
It appears that the junit runner main method will call system.exist and force the process to terminate even when a non daemon thread is still running.

    public static void main(String... args) {
        Result result = new JUnitCore().runMain(new RealSystem(), args);
        System.exit(result.wasSuccessful() ? 0 : 1);
    }